pkg/restclient/common.go

```diff
@@ -21,17 +19,111 @@ const (
 	defaultClientTimeout = 30 // in seconds
 )
 
+// ClientFunc receives a Client and overrides its members
+type ClientFunc func(*Client) error
+
+// WithTLSConfig overrides http.Client member with a client created
+// using specified TLS configuration.
+func WithTLSConfig(tlsOpts *TLSOptions) ClientFunc {
+	return func(client *Client) error {
+		tlsConfig, err := NewTLSConfig(tlsOpts)
+		if err != nil {
+			return err
+		}
+		tr := &http.Transport{
+			DisableCompression: true,
+			DisableKeepAlives:  true,
+			TLSClientConfig:    tlsConfig,
+		}
+		httpClient := &http.Client{
+			Timeout:   defaultClientTimeout * time.Second,
```

Comment:
setting `Timeout` can be moved to `NewClientWithOpts` function

Reply:
this is default client time out, we can modify it later using WithTimeOut(..)